src/main/java/com/aws/greengrass/deployment/exceptions/RetryableCloudDeploymentException.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.deployment.exceptions;
+
+import com.aws.greengrass.deployment.errorcode.DeploymentErrorCode;
+
+/**
+ * Exception for handling 5xx deployment failures.
+ */
+public class RetryableCloudDeploymentException extends DeploymentException {
```

Comment:
I asked this somewhere else too before but couldn't find an answer. Why are we specifically calling this `RetryableCloudDeploymentException` error? are retries only relevant for cloud deployment? can a local deployment not include API calls to GG cloud/other dependencies like S3 when the device is offline? If it is possible for local deployments to hit this code path, we should decide whether - 1) we want retries for local deployment or not 2) if we do want retries for local deployments the naming shouldn't mention cloud specifically. If we don't want retries for local deployments we should to test that behavior for local deployments